### PR TITLE
Bugfix/cam fixes 5

### PR DESCRIPTION
**Pull request by MarcelCaron-NOAA**
<b>Note to developers: You must use this PR template!</b>

## Description of Changes

> Please include a summary of the changes and the related GitHub issue(s). Please also include relevant motivation and context.

- _Bugfix:_ Bad lines in stats files (rare) cause fatal errors when plotting.  _Outcome_:  Print a warning and then skip bad lines.
- _Bugfix:_ Key name for accessing netCDF precip map data is incorrect.  _Outcome_:  Update key, with back-up options if that key fails.
- _Bugfix:_ Runs hit memory limit, depending on cycle.  _Outcome_:  Memory ceiling should encapsulate largest runs of the same job.
- _Bugfix:_ Since #597 was merged, "gather3" job card commands were being muted for precip and snowfall stats jobs due to a missing subdirectory when checking for needed files, and no final stats files were being made. _Outcome_: Final stats should now be written.

## Developer Questions and Checklist
* Is this a high priority PR? If so, why and is there a date it needs to be merged by?

No
* Do you have any planned upcoming annual leave/PTO?

Yes: 12/24-25, 12/30-1/3
* Are there any changes needed for when the jobs are supposed to run?

No
  
- [x] The code changes follow [NCO's EE2 Standards](https://www.nco.ncep.noaa.gov/idsb/implementation_standards/ImplementationStandards.v11.0.0.pdf).
- [x] Developer's name is removed throughout the code and have used `${USER}` where necessary throughout the code.
- [x] References the feature branch for `HOMEevs` are removed from the code.
- [x] J-Job environment variables, COMIN and COMOUT directories, and output follow what has been [defined](https://docs.google.com/document/d/1JWg_4q80aYmmAoD21GFjp9R9y5-3w7WGM3-0HJk0Pjs/edit#heading=h.7ysbr191vzu4) for EVS.
- [x] Jobs over 15 minutes in runtime have restart capability.
- [x] If applicable, changes in the `dev/drivers/scripts` or `dev/modulefiles` have been made in the corresponding `ecf/scripts` and `ecf/defs/evs-nco.def`? 
- [x] Jobs contain the approriate file checking and don't run METplus for any missing data.
- [x] Code is using METplus wrappers structure and not calling MET executables directly.
- [x] Log is free of any ERRORs or WARNINGs.

## Testing Instructions

> Please include testing instructions for the PR assignee. Include all relevant input datasets needed to run the tests.

#### (1) Set up jobs

- symlink the EVS_fix directory locally as "fix"
- In each driver script, edit the following environment variables:
**HOMEevs** - set to your test EVS directory
**COMIN** - set to `/lfs/h2/emc/vpppg/noscrub/emc.vpppg/${NET}/$evs_ver_2d`
**COMOUT** - set to your test output directory
**SENDMAIL** - (_optional_) set to `"NO"`
**DATAROOT** - (_optional_) set to your test DATAROOT directory

#### (2) Running jobs

I recommend testing the following jobs:
```
jevs_cam_hireswarw_severe_prep
jevs_cam_hireswarwmem2_severe_prep
jevs_cam_hireswfv3_severe_prep
jevs_cam_hrrr_severe_prep
jevs_cam_namnest_severe_prep
```
```
jevs_cam_namnest_precip_stats
```
```
jevs_cam_precip_plots_last31days
```
[Total: 1 prep job; 1 stats job; 1 plots job]

#### (3) Testing jobs

- The prep driver scripts should be submitted using `qsub -v vhr=12 $driver`
- The plots driver script can be submitted using `qsub -v vhr=00 $driver`

For stats job, please run with three different values for `vhr`:
- `qsub -v vhr=19 $driver`
- `qsub -v vhr=20 $driver`
- `qsub -v vhr=21 $driver`
- _Note_:  These can be run concurrently, since the job takes a while to complete.

#### (4) Checking jobs
- Log files should be checked by the developer for the following keywords:
```
check="FATAL\|WARNING\|error\|Error\|Killed\|Cgroup\|argument expected\|No such file\|cannot\|failed\|unexpected\|exceeded"
grep "$check" $logfile
```

### CODE REVIEW

**Comment by malloryprow** [comment]
### 1. jevs_cam_hrrr_severe_prep (vhr=12)
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr618/EVS/dev/drivers/scripts/prep/cam/jevs_cam_hrrr_severe_prep.o209004872 
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hrrr_severe_prep.209004872.dbqs01
**COMOUT:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr618/evs/v2.0/prep/cam

**Comment by malloryprow** [comment]
COMOUT is  /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr618/evs/v2.0/prep/cam

### 1. jevs_cam_namnest_precip_stats (vhr=12)
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr618/EVS/dev/drivers/scripts/stats/cam/jevs_cam_namnest_precip_stats.o209005509
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_namnest_precip_stats.209005509.dbqs01

### 2. jevs_cam_namnest_precip_stats (vhr=20)
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr618/EVS/dev/drivers/scripts/stats/cam/jevs_cam_namnest_precip_stats.o209005583
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_namnest_precip_stats.209005583.dbqs01

### 3. jevs_cam_namnest_precip_stats (vhr=21)
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr618/EVS/dev/drivers/scripts/stats/cam/jevs_cam_namnest_precip_stats.o209005596
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_namnest_precip_stats.209005596.dbqs01

**Comment by malloryprow** [comment]
### 1. jevs_cam_precip_plots_last31days (vhr=00)
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr618/EVS/dev/drivers/scripts/plots/cam/jevs_cam_precip_plots_last31days.o209005682
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_precip_plots_last31days.209005682.dbqs01
**COMOUT:**/lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr618/evs/v2.0/plots/cam

**Comment by malloryprow** [comment]
And if you could do a sync fork when you get a chance too!

File: dev/drivers/scripts/prep/cam/jevs_cam_hrrr_severe_prep.sh [hunk 1]
```diff
@@ -4,7 +4,7 @@
 #PBS -q dev
 #PBS -A VERF-DEV
 #PBS -l walltime=0:30:00
```
Comment (malloryprow) [inline dev/drivers/scripts/prep/cam/jevs_cam_hrrr_severe_prep.sh]: Job ran for ~1 min in my testing and I see in the parallel the max runtime it has is also ~1 min. Can we make the walltime 10 min to follow NCO's guideline for jobs run 0-2 min to have a walltime of 10 min?

File: ecf/scripts/prep/cam/jevs_cam_hrrr_severe_prep_vhr_master.ecf [hunk 2]
```diff
@@ -4,7 +4,7 @@
 #PBS -q %QUEUE%
 #PBS -A %PROJ%-%PROJENVIR%
 #PBS -l walltime=00:10:00
```
Comment (malloryprow) [inline ecf/scripts/prep/cam/jevs_cam_hrrr_severe_prep_vhr_master.ecf:6]: Job ran for ~1 min in my testing and I see in the parallel the max runtime it has is also ~1 min. Can we make the walltime 10 min to follow NCO's guideline for jobs run 0-2 min to have a walltime of 10 min?

File: dev/drivers/scripts/prep/cam/jevs_cam_hrrr_severe_prep.sh (hunk 1 above)
Comment (MarcelCaron-NOAA) [reply dev/drivers/scripts/prep/cam/jevs_cam_hrrr_severe_prep.sh]: Sure, let me change that now

Comment (MarcelCaron-NOAA) [reply dev/drivers/scripts/prep/cam/jevs_cam_hrrr_severe_prep.sh]: Done 👍

File: ecf/scripts/prep/cam/jevs_cam_hrrr_severe_prep_vhr_master.ecf (hunk 2 above)
Comment (MarcelCaron-NOAA) [reply ecf/scripts/prep/cam/jevs_cam_hrrr_severe_prep_vhr_master.ecf:6]: Looks like it already is 10 minutes.  I edited other severe jobs to match their ecf scripts.  Maybe we can run those too?
- jevs_cam_hireswarw_severe_prep
- jevs_cam_hireswarwmem2_severe_prep
- jevs_cam_hireswfv3_severe_prep
- jevs_cam_namnest_severe_prep

Comment (malloryprow) [reply ecf/scripts/prep/cam/jevs_cam_hrrr_severe_prep_vhr_master.ecf:6]: Sure! Should I do `qsub -v vhr=12` to submit them?

**Comment by MarcelCaron-NOAA** [comment]
I was wondering why precip plots finished so quickly.  Looks like a bug was introduced in PR #597 causing final gather job cards to be muted.  I pushed the fix to the current PR [(commit)](https://github.com/NOAA-EMC/EVS/pull/618/commits/7ce432e801350080fc2080ef4292efbdd45434f9) but there is no final **precip** or **snowfall** data in the parallel EVS since 11/10/2024.  If we want to backfill the archive, the small stats files are in COMOUT and used to run the final gather jobs manually and retrospectively for 11/11-present.  Not sure how we want to do that.

❌ jevs_cam_precip_plots_last31days

In any case, can we rerun **jevs_cam_precip_plots_last31days** like this:
```
qsub -v vhr=00,VDATE=20241110 $driver_dir/jevs_cam_precip_plots_last31days.sh
```

**Comment by MarcelCaron-NOAA** [comment]
The following all completed successfully and output look normal:
✔️ jevs_cam_hrrr_severe_prep
✔️ jevs_cam_namnest_precip_stats (all runs)

👍 No other concerns from me about **stats**

File: ecf/scripts/prep/cam/jevs_cam_hrrr_severe_prep_vhr_master.ecf (hunk 2 above)
Comment (MarcelCaron-NOAA) [reply ecf/scripts/prep/cam/jevs_cam_hrrr_severe_prep_vhr_master.ecf:6]: Yes, thanks!

**Comment by malloryprow** [comment]
### 1. jevs_cam_precip_plots_last31days (vhr=00, VDATE=20241110)
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr618/EVS/dev/drivers/scripts/plots/cam/jevs_cam_precip_plots_last31days.o209010211
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_precip_plots_last31days.209010211.dbqs01
**COMOUT:**/lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr618/evs/v2.0/plots/cam

**Comment by malloryprow** [comment]
### 1. jevs_cam_hireswarw_severe_prep (vhr=12)
**Log File:**/lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr618/EVS/dev/drivers/scripts/prep/cam/jevs_cam_hireswarw_severe_prep.o209010806
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hireswarw_severe_prep.209010806.dbqs01

### 2. jevs_cam_hireswarwmem2_severe_prep (vhr=12)
**Log File:**/lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr618/EVS/dev/drivers/scripts/prep/cam/jevs_cam_hireswarwmem2_severe_prep.o209010855
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hireswarwmem2_severe_prep.209010855.dbqs01

### 3. jevs_cam_hireswfv3_severe_prep (vhr=12)
**Log File:**/lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr618/EVS/dev/drivers/scripts/prep/cam/jevs_cam_hireswfv3_severe_prep.o209010890
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hireswfv3_severe_prep.209010890.dbqs01

### 4. jevs_cam_namnest_severe_prep (vhr=12)
**Log File:**/lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr618/EVS/dev/drivers/scripts/prep/cam/jevs_cam_namnest_severe_prep.o209010897
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_namnest_severe_prep.209010897.dbqs01

**Comment by MarcelCaron-NOAA** [comment]
The following completed successfully and output files look normal:
✔️ jevs_cam_hireswarw_severe_prep
✔️ jevs_cam_hireswarwmem2_severe_prep
✔️ jevs_cam_hireswfv3_severe_prep
✔️ jevs_cam_namnest_severe_prep

👍 No more concerns from me about **prep**

The following is still running:
⏳ jevs_cam_precip_plots_last31days

**Comment by MarcelCaron-NOAA** [comment]
✔️ jevs_cam_precip_plots_last31days completed cleanly and output file looks good

👍 No other concerns from me about the changes in this PR.

**Review by malloryprow — APPROVED** [review]
Changes are good and testing successful.

Thanks for your work on this @MarcelCaron-NOAA! Cross off anything on the Fixes and Additions document.

**Review by AndrewBenjamin-NOAA — APPROVED** [review]
I have reviewed these changes and approve this PR. Thank you for the continued cam bugfixes!

**Pull request merged**